sklearn/calibration.py

```diff
@@ -303,15 +305,13 @@ def fit(self, X, y, sample_weight=None):
             # sample_weight checks
             fit_parameters = signature(base_estimator.fit).parameters
             supports_sw = "sample_weight" in fit_parameters
-            if sample_weight is not None:
-                sample_weight = _check_sample_weight(sample_weight, X)
-                if not supports_sw:
-                    estimator_name = type(base_estimator).__name__
-                    warnings.warn(
-                        f"Since {estimator_name} does not support "
-                        "sample_weights, sample weights will only be"
-                        " used for the calibration itself."
-                    )
+            if sample_weight is not None and not supports_sw:
+                estimator_name = type(base_estimator).__name__
+                warnings.warn(
+                    f"Since {estimator_name} does not support "
+                    "sample_weights, sample weights will only be"
+                    " used for the calibration itself."
```

Comment:
Maybe we could be more explicit and say that this is a known bug of scikit-learn and link to an issue to track it.